KIPs/kip-81.md

```diff
@@ -0,0 +1,335 @@
+| `execDelay`       | A minimum delay before a queued transaction can be executed.                              | 2 days    |
+| `execTimeout`     | Grace period given to executors for execute() queued proposals after `execDelay`.         | 7 days    |
+
+For each proposal, several important block numbers are determined.
+
+| Name              | Meaning                                                                                                                                                                   |
+|-----------------  |-------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `proposedBlock`   | The block number where `propose()` was called.                                                                                                                            |
+| `startBlock`      | Voting start block `proposedBlock + votingDelay`.                                                                                                                         |
+| `endBlock`        | Voting end block `startBlock + votingPeriod`. An executor should queue transactions in the proposal before `endBlock + queueTimeout`, otherwise the proposal expires.     |
+| `queuedBlock`     | The block number where `queue()` was called. An executor should trigger the transaction after `queuedBlock + execDelay`, before `queuedBlock + execDelay + execTimeout`.  |
+
+The timings are illustrated in below diagram.
+
+![proposal timing](../assets/kip-81/proposal_timing.png)
+
+A proposal is in one of the following states.
+
+- **Pending**: Proposer submitted the proposal but voting has not started. The pending state lasts for `votingDelay`. This state is also called the **notice period**.
+- **Active**: Ongoing voting. Voters can cast their votes. The active state lasts for `votingPeriod`. This state is also called the **voting period**.
+- **Canceled**: Proposer has canceled the proposal before execution. Any further interaction is prohibited.
+- **Passed**: Voting finalized and quorum reached. An executor shall queue the actions before a set timeout of `queueTimeout`. If the proposal contains no actions, any further interaction is prohibited. Otherwise, the proposal can proceed to Queued and Executed states.
+- **Failed**: Voting finalized but has not reached quorum. Any further interaction is prohibited.
+- **Queued**: An executor has queued the actions. An executor cannot trigger the actions while a certain duration of `execDelay`.
+- **Executed**: An executor has triggered the actions after the execution delay. The executor shall trigger the actions before a set timeout of `execTimeout`. The `execTimeout` starts at the end of `execDelay`.
+- **Expired**: Executor has not queued or executed within respective timeouts.
```

Comment:
Is there any penalty if the executor does not execute the transactions?

Reply:
No there is no penalty for not executing the transactions.